sdk/logs/src/test/java/io/opentelemetry/sdk/logs/LoggerConfigTest.java

```diff
@@ -0,0 +1,126 @@
+    loggerConfig = loggerConfig == null ? defaultConfig() : loggerConfig;
+    assertThat(loggerConfig).isEqualTo(expectedLoggerConfig);
+  }
+
+  private static final InstrumentationScopeInfo scopeCat = InstrumentationScopeInfo.create("cat");
+  private static final InstrumentationScopeInfo scopeDog = InstrumentationScopeInfo.create("dog");
+  private static final InstrumentationScopeInfo scopeDuck = InstrumentationScopeInfo.create("duck");
+
+  private static Stream<Arguments> loggerConfiguratorArgs() {
+    ScopeConfigurator<LoggerConfig> defaultConfigurator =
+        LoggerConfig.configuratorBuilder().build();
+    ScopeConfigurator<LoggerConfig> disableCat =
+        LoggerConfig.configuratorBuilder()
+            .addCondition(nameEquals("cat"), LoggerConfig.disabled())
+            // Second matching rule for cat should be ignored
+            .addCondition(nameEquals("cat"), enabled())
+            .build();
+    ScopeConfigurator<LoggerConfig> disableStartsWithD =
+        LoggerConfig.configuratorBuilder()
+            .addCondition(nameMatchesGlob("d*"), LoggerConfig.disabled())
+            .build();
+    ScopeConfigurator<LoggerConfig> enableCat =
+        LoggerConfig.configuratorBuilder()
+            .setDefault(LoggerConfig.disabled())
+            .addCondition(nameEquals("cat"), enabled())
+            // Second matching rule for cat should be ignored
+            .addCondition(nameEquals("cat"), LoggerConfig.disabled())
+            .build();
+    ScopeConfigurator<LoggerConfig> enableStartsWithD =
+        LoggerConfig.configuratorBuilder()
+            .setDefault(LoggerConfig.disabled())
+            .addCondition(nameMatchesGlob("d*"), LoggerConfig.enabled())
+            .build();
+
+    return Stream.of(
+        // default
+        Arguments.of(defaultConfigurator, scopeCat, defaultConfig()),
+        Arguments.of(defaultConfigurator, scopeDog, defaultConfig()),
+        Arguments.of(defaultConfigurator, scopeDuck, defaultConfig()),
+        // default enabled, disable cat
+        Arguments.of(disableCat, scopeCat, LoggerConfig.disabled()),
+        Arguments.of(disableCat, scopeDog, enabled()),
+        Arguments.of(disableCat, scopeDuck, enabled()),
+        // default enabled, disable pattern
+        Arguments.of(disableStartsWithD, scopeCat, enabled()),
+        Arguments.of(disableStartsWithD, scopeDog, LoggerConfig.disabled()),
+        Arguments.of(disableStartsWithD, scopeDuck, LoggerConfig.disabled()),
+        // default disabled, enable cat
+        Arguments.of(enableCat, scopeCat, enabled()),
+        Arguments.of(enableCat, scopeDog, LoggerConfig.disabled()),
+        Arguments.of(enableCat, scopeDuck, LoggerConfig.disabled()),
+        // default disabled, enable pattern
+        Arguments.of(enableStartsWithD, scopeCat, LoggerConfig.disabled()),
+        Arguments.of(enableStartsWithD, scopeDog, enabled()),
+        Arguments.of(enableStartsWithD, scopeDuck, enabled()));
```

Comment:
Took me a second to think about, but I'm loving this parameterized coverage.

Reply:
Yeah I picked this up from go, where this type of parameterized test is extremely common. Little more effort to understand, but once you do, it massively increases the information density by removing a bunch of repeated boilerplate we'd normally see.